src/filters/mentions.cjs

```diff
@@ -1,92 +1,72 @@
 'use strict';
-const _ = require('lodash');
+
 const sanitizeHTML = require('sanitize-html');
+const truncate = require('truncate-html');
+
+// define which types of webmentions are included by default
+// possible values listed here:
+// https://github.com/aaronpk/webmention.io#find-links-of-a-specific-type-to-a-specific-page
+const defaultTypes = ['mention-of', 'in-reply-to'];
 
 // sort webmentions by published timestamp chronologically.
 // swap a.published and b.published to reverse order.
-const orderByDate = (a, b) => new Date(b.published) - new Date(a.published);
-
-const forUrl = (webMentions, url) =>
-  webMentions.children
-    .filter((entry) => entry['wm-target'] === url)
-    .filter((entry) => entry['wm-private'] === false)
-    .sort(orderByDate);
-
-const authors = (mentions) => {
-  const photos = mentions
-    .map((entry) => {
-      entry.author.property = entry['wm-property'];
-      return entry.author;
-    })
-    .filter((author) => author.photo);
+const orderByDate = (a, b) => new Date(a.published) - new Date(b.published);
 
-  return _.uniqBy(photos, 'photo').map((a) => {
-    a.property = _.uniq(
-      _.filter(photos, ['photo', a.photo]).map((i) => i.property),
-    );
-    return a;
-  });
+const normalizeURL = (input) => {
+  const url = new URL(input);
+  url.search = url.hash = '';
+  const output = url.toString();
+  if (!output.endsWith('/')) {
+    return `${output}/`;
+  }
+  return output;
 };
 
-const likes = (mentions) =>
-  mentions.filter((entry) => entry['wm-property'] === 'like-of');
-
-const reposts = (mentions) =>
-  mentions.filter((entry) => entry['wm-property'] === 'repost-of');
-
-const webMentions = (mentions) => {
-  // define which types of webmentions should be included per URL.
-  // possible values listed here:
-  // https://github.com/aaronpk/webmention.io#find-links-of-a-specific-type-to-a-specific-page
-  const allowedTypes = ['mention-of', 'in-reply-to', 'bookmark-of'];
+const forUrl = (mentions, url) =>
+  mentions.children
+    .filter(
+      (entry) =>
+        normalizeURL(entry['wm-target']) === normalizeURL(url) &&
+        !entry['wm-private'],
+    )
+    .sort(orderByDate);
 
-  // clean webmention content for output
-  const clean = (entry) => {
-    const { html, text } = entry.content;
+// clean webmention content for output
+const clean = (entry) => {
+  const { html, text } = entry.content;
 
-    if (html) {
-      // really long html mentions, usually newsletters or compilations
-      entry.content.value = sanitizeHTML(
-        html.length > 2000
-          ? `
-            <p>${html.substring(0, 250)}…</p>
-            <p>
-              Read more:
-              <a href="${entry['wm-source']}">${entry['wm-source']}</a>
-            </p>
-          `
-          : html,
-      );
-    } else {
-      entry.content.value = sanitizeHTML(text);
+  if (html) {
+    let truncated = truncate(html, 50, { byWords: true, ellipsis: '…' });
```

Comment:
@mirisuzanne This is now using a more consistent way to truncate the HTML, but it makes it difficult to know for sure when something has been truncated, so I removed the "Read more" addition. I think that's unnecessary anyway, when the title is already a link?